database/database.go

```diff
@@ -1,127 +1,585 @@
-// sudo docker run --name some-postgres -e POSTGRES_PASSWORD=1234 -e POSTGRES_DB=quote_gallery -p 0.0.0.0:5432:5432 -d postgres
-
 package database
 
 import (
 	"database/sql"
+	"errors"
 	"log"
 
 	// loading postgresql driver
 	_ "github.com/lib/pq"
 )
 
-// Teacher struct
-type Teacher struct {
-	ID    int
-	Name  string
-	Title string
-	Note  string
+/* -------------------------------------------------------------------------- */
+/*                                 DEFINITIONS                                */
+/* -------------------------------------------------------------------------- */
+
+// QuoteT stores one quote
+// QuoteID   the unique identificator of the quote
+// TeacherID the unique identifier of the corresponding teacher
+// Context	  the context of the quote
+// text       the text of the quote itself
+// unixtime   optional
+//
+// match  	  only locally, not safed in PostgreSQL database!
+//			  used by GetQuotesFromString to quantify how well this quote fits the string
+type QuoteT struct {
+	QuoteID   uint32
+	TeacherID uint32
+	Context   string
+	Text      string
+	Unixtime  uint64
+	Upvotes   uint32
+	Match     float32
 }
 
-// Quote struct
-type Quote struct {
-	ID      int
-	Teacher Teacher
-	Text    string
+// UnverifiedQuoteT stores one unverified quote
+// QuoteID    the unique identificator of the unverified quote
+// teacher 	   the name of the teacher
+// Context	   the context of the quote
+// text        the text of the quote itself
+// unixtime    optional
+// iphash	   optional
+//
+type UnverifiedQuoteT struct {
+	QuoteID  uint32
+	Teacher  string
+	Context  string
+	Text     string
+	Unixtime uint64
+	IPHash   uint64
 }
 
-var database *sql.DB
+// TeacherT stores one teacher
+// TeacherID the unique identifier of the teacher
+// name       the teacher's name
+// title      the teacher's title
+// note       optional notes, e.g. subjects
+type TeacherT struct {
+	TeacherID uint32
+	Name      string
+	Title     string
+	Note      string
+}
+
+type wordsMapT struct {
+	totalOccurences uint32
+	occurenceSlice  []occurenceSliceT
+}
+
+type occurenceSliceT struct {
+	enumid int32
+	count  uint32
+}
+
+/* -------------------------------------------------------------------------- */
+/*                          GLOBAL PACKAGE VARIABLES                          */
+/* -------------------------------------------------------------------------- */
+
+// Handle to the PostgreSQL database, used as long time storage
+var postgresDatabase *sql.DB
 
-// SetupDatabase is a function to setup the database
-func SetupDatabase() {
+// Created from PostgreSQL database at (re)start
+// localDatabase is a cache of the postgreSQL database to speed up read operations
+//
+// unverified quotes will not be cached in the local database, because read operations
+// will only be performed by the operator and thus be very rare
+//
+// important: the unverifiedQuote table will not be mirrored
```

Comment:
you can combine these two statements
```suggestion
// unverified quotes (the unverifiedQuote table) will not be cached because read operations
// will only be performed by the operator and thus be very rare
```
btw +1 for fine English